src/entry.rs

```diff
@@ -475,6 +475,23 @@ impl<'a> EntryFields<'a> {
                     // links though they're canonicalized to their existing path
                     // so we need to validate at this time.
                     Some(ref p) => {
+                        let src = if src.is_absolute() {
+                            // link points to an absolute path, assume it's relative to the target dir
+                            // by removing the leading '/' as we already do for files, see unpack_in() comments.
```

Comment:
Could this update the commnt above? It seems like that doc block should probably be documenting what's going on here.

 Could this perhaps be structured also a bit differently to do `src.to_path_buf()` early on and only mutate that binding if the condition to trim it is true?

Reply:
Sure, what about this?